lib/index.ts

```diff
@@ -85,10 +85,38 @@ class PWMetrics {
         const sheets = new Sheets(this.sheets, this.clientSecret);
         await sheets.appendResults(results.runs);
       }
+
+      if (this.resultHasExpectationErrors(results) && this.flags.expectations) {
+        throw new Error(messages.getMessage('HAS_EXPECTATION_ERRORS'));
+      }
     }
     return results;
   }
 
+  resultHasExpectationErrors(results: PWMetricsResults): boolean {
+    const expectationErrors = results.runs.filter(run => {
+      return run.timings.filter(item => {
```

Comment:
I totally agree that's pretty heavy do that, but `checkExpectations` uses `Timing` interface and add some flag there creates a cascade step if I understood correctl (PWMetricsResults >> MetricsResults >> Timing). Could you elaborate a little bit more about it, please?